verticles/graphql/src/main/java/com/gentics/mesh/graphql/model/NodeReferenceIn.java

```diff
@@ -58,6 +66,43 @@ public static Stream<NodeReferenceIn> fromContent(GraphQLContext gc, NodeContent
 						ref)));
 	}
 
+	/**
+	 * Creates a stream of incoming node references for the given content.
+	 * @param gc
+	 * @param contentTypes a content/type pair
+	 * @return
+	 */
+	public static Stream<Pair<NodeReferenceIn, NodeContent>> fromContent(GraphQLContext gc, Collection<NodeContent> contents) {
+		Tx tx = Tx.get();
+		ContentDao contentDao = tx.contentDao();
+		String branchUuid = tx.getBranch(gc).getUuid();
+
+		// content per node
+		Map<HibNode, List<NodeContent>> nodeContent = contents.stream().collect(Collectors.groupingBy(NodeContent::getNode, Collectors.mapping(Function.identity(), Collectors.toList())));
+
+		// 1. Collecting referencing fields for the nodes
+		return contentDao.getInboundReferences(
+					contents.stream().map(NodeContent::getNode).collect(Collectors.toSet())
+				// 2. Collect the contents of the referencing fields
+				).flatMap(refNode -> refNode.getKey().getReferencingContents()
```

Comment:
This will still use extra queries per node.

Reply:
I tested the current implementation by loading "referencedBy" for 10 nodes, where each one is referenced by 10 other nodes. The REFERENCED_BY_LOADER used over 600 SELECT Statements for that.

Reply:
Actual for Enterprise only -> goes into Enterprise part.